src/org/purefn/sqlium/import.clj

```diff
@@ -39,8 +40,14 @@
   [db sql-with-aliases]
   (let [[sql col-aliases] sql-with-aliases]
     ;; TODO: fix jdbc call
-    (->> (jdbc/query db [sql])
-         (map #(set/rename-keys % col-aliases)))))
+    (try
+      (->> (jdbc/query db [sql])
+           (map #(set/rename-keys % col-aliases)))
+      ;TODO: possible improvement
+      ;- catch specific error codes only
+      ;- toggle try-catch on/off based on a config flag
+      (catch SQLException e
+        (log/error "Error querying database:  " (.getMessage e) "\n" e)))))
```

Comment:
I think a better way to log this would be:

`(log/error e "Error querying database" {:query sql})`

That'll properly format the stack trace, as well as log the exact query that caused it.

Reply:
I don't really think error handling belongs in this library as such, receiving a `nil` on failure seems brittle.

If I'm familiar with what we're trying to address, why not solve the problem in the general case in `sentenza`
https://github.com/PureFnOrg/sentenza/blob/dd72667bcf08018fb4a5eae0d91eb3eac285b3f4/src/org/purefn/sentenza/api.clj#L67 
or
https://github.com/PureFnOrg/sentenza/blob/dd72667bcf08018fb4a5eae0d91eb3eac285b3f4/src/org/purefn/sentenza/api.clj#L63

Reply:
I'm not sure I agree that error handling for the `source` in `sentenza` doesn't fit.  It seems reasonable to build a pipeline which doesn't halt computation when whatever mechanism is producing a source element throws an exception.

I don't follow point 2, couldn't this:
https://github.com/PureFnOrg/sentenza/blob/dd72667bcf08018fb4a5eae0d91eb3eac285b3f4/src/org/purefn/sentenza/api.clj#L67-L69
be something closer to `core.async/onto-chan` with error handling?  The expectation here is that the `fn` returns a `lazy-seq`

Reply:
I think a general behavior of catching exceptions so pipeline completes may not be ideal since sentenza use-cases can differ.
Perhaps it makes sense as a user configurable behavior?

Regarding 2, if I understand you suggestion correctly, the idea is to catch exception when individual contents of the returned lazy-seq get realized right?
The issue I see is clojure seems to do realization in chunks. [chuncked-realization](http://clojure-doc.org/articles/language/laziness.html#lazy-sequences-chunking)
Thus, the handling we do (in this case, skipping) will end up affecting the whole chunk.

Reply:
Actually if you get an exception on realizing lazy seq, that lazy seq is finished. You're not getting anything after that.

I thought more about what this PR is doing, and it's going to have even weirder and more unpredictable effects than I originally realized.

The `fetch-results` function is actually called potentially multiple times for each batch of records, depending on the complexity of the entity spec. Each one-to-many relationship in the spec will result in an additional query, and additional call to this function. Depending on which one fails and nils out, several outcomes are possible:

- If it fails on the initial query to get the entity root, then batch-size # of records (default batch size is 10,000) will silently be missing.
- If it fails on a query to get a one-to-many relationship, all that data and any one-to-many relationships recursively underneath will be silently missing for the entire batch.

Silently missing random data is not an appropriate error handling mechanism.

Reply:
@YTN01 is that piece of documentation correct?  I'd expect to see more side effects printing if the `lazy-seq` was being realized in chunks.  Are chunked lazy seqs opt-in?

https://repl.it/@MattKocubinski/DecimalMinorDeclarations

Also @mgaare (from the same REPL link) this is naive retry, but couldn't this strategy work for re-attempting a failed thunk during `lazy-seq` realization?

Reply:
@kocubinski yeah i tried it and was able to confirm 32 items get realized at a time.

Reply:
@YTN01 Why do say that? Doesn't the REPL snippet I posted disprove that?

Reply:
@kocubinski not sure why it didn't happen there.
But you can see it here: https://repl.it/repls/FewMonthlyArchitects

Reply:
@YTN01 Got it.  So it does look like `Range` implements chunking, which can be seen here - https://github.com/clojure/clojure/blob/clojure-1.9.0/src/jvm/clojure/lang/Range.java 
However the implementation for the `lazy-seq` macro does not realize in chunks.

Reply:
@kocubinski My understanding is `lazy-seq` is a wrapper over other seq'uable types, and its behavior for operations like `first` or `take` would depend on the respective behavior of the underlying collection type.
e.g, for a range wrapped by a lazy-seq, chunking would still happen on read calls.